// react-gravatar/dist/index.js
// Generated by CoffeeScript 1.9.1
var React, isRetina, md5, querystring;

React = require('react');

md5 = require('md5');

querystring = require('querystring');

isRetina = require('is-retina');

module.exports = React.createClass({
  displayName: 'Gravatar',
  propTypes: {
    email: React.PropTypes.string.isRequired,
    size: React.PropTypes.number,
    rating: React.PropTypes.string,
    https: React.PropTypes.bool,
    "default": React.PropTypes.string,
    className: React.PropTypes.string
  },
  getDefaultProps: function() {
    return {
      size: 50,
      rating: 'g',
      https: false,
      "default": "retro",
      email: '',
      className: ""
    };
  },
  render: function() {
    var base, query, src;
    base = this.props.https ? "https://secure.gravatar.com/avatar/" : 'http://www.gravatar.com/avatar/';
    query = querystring.stringify({
      s: isRetina() ? this.props.size * 2 : this.props.size,
      r: this.props.rating,
      d: this.props["default"]
    });
    src = base + md5.digest_s(this.props.email) + "?" + query;
    return React.createElement("img", React.__spread({}, this.props, {
      "className": "react-gravatar " + this.props.className,
      "src": src,
      "alt": this.props.email,
      "height": this.props.size,
      "width": this.props.size
    }));
  }
});
